### ISSUE TITLE
[Wasm][AOT] memory access out of bounds in mini_llvmonly_get_imt_trampoline 

### ISSUE THREAD

**Issue by jeromelaban** [dotnet/runtime#48814]
### Description

The following runtime failure happens when running in FullAOT or Mixed AOT (without profile) :

```
Unhandled dispatcher exception: RuntimeError: memory access out of bounds (RuntimeError: memory access out of bounds
    at m_class_get_image.28 (<anonymous>:wasm-function[126941]:0x2c5b756)
    at m_class_get_mem_manager.3 (<anonymous>:wasm-function[126919]:0x2c5ae31)
    at m_class_alloc.1 (<anonymous>:wasm-function[126911]:0x2c5aa76)
    at mini_llvmonly_get_imt_trampoline (<anonymous>:wasm-function[126909]:0x2c5a8a7)
    at mono_method_add_generic_virtual_invocation (<anonymous>:wasm-function[123258]:0x2be3e28)
    at mini_llvmonly_resolve_generic_virtual_call (<anonymous>:wasm-function[126927]:0x2c5b0be)
    at System_Linq_System_Linq_Enumerable_Select_TSource_REF_TResult_REF_System_Collections_Generic_IEnumerable_1_TSource_REF_System_Func_2_TSource_REF_TResult_REF (<anonymous>:wasm-function[25631]:0x8aba6f)
    at Uno_UI_Windows_UI_Xaml_DependencyObjectExtensions_RegisterDisposableNestedPropertyChangedCallback_Windows_UI_Xaml_DependencyObject_Windows_UI_Xaml_PropertyChangedCallback_Windows_UI_Xaml_DependencyProperty____ (<anonymous>:wasm-function[56200]:0x1408a65)
    at Uno_UI_Windows_UI_Xaml_UIElement_OnClipChanged_Windows_UI_Xaml_DependencyPropertyChangedEventArgs (<anonymous>:wasm-function[56194]:0x140826e)
    at Uno_UI_Windows_UI_Xaml_UIElement__c___cctorb__395_0_Windows_UI_Xaml_DependencyObject_Windows_UI_Xaml_DependencyPropertyChangedEventArgs (<anonymous>:wasm-function[56688]:0x142cb36))
```

Repro not yet available.

### Configuration

e912151eda83eb13b41896fe480cd6f5f1465af2

### Regression?

No

**Comment by jeromelaban** [comment]
This issue can be reproduced by enabling dedup generation

**Comment by jeromelaban** [comment]
This is a regression from https://github.com/dotnet/runtime/pull/48818

**Comment by lewing** [comment]
There have been some memory manager related changes, have you tried this recently?

**Comment by jeromelaban** [comment]
As with all problems of this kind, it only happens on very specific scenarios, so I was not able to create a repro. I hope that https://github.com/dotnet/runtime/issues/51207 will help track one down as it is 100% reproducible with the sample I provided.

**Comment by jeromelaban** [comment]
This issue was fixed alongside #51207.
